src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -306,5 +309,112 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
+        {
+            using (FileStream csvStream = new FileStream(path, FileMode.Create))
+            {
+                WriteCsv(dataFrame: dataFrame, csvStream: csvStream,
+                           separator: separator, header: header,
+                           encoding: encoding, cultureInfo: cultureInfo);
+            }
+        }
+
+        /// <summary>
+        /// Writes a DataFrame into a CSV.
+        /// Follows pandas API.
+        /// </summary>
+        /// <param name="dataFrame"><see cref="DataFrame"/></param>
+        /// <param name="csvStream">stream of CSV data to be write out</param>
+        /// <param name="separator">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="encoding">the character encoding. Defaults to UTF8 if not specified</param>
+        /// <param name="cultureInfo">culture info for handeling decimal values</param>
+        public static void WriteCsv(DataFrame dataFrame, Stream csvStream,
+                           char separator = ',', bool header = true,
+                           Encoding encoding = null, CultureInfo cultureInfo = null)
+        {
+            if (cultureInfo is null)
+            {
+                cultureInfo = CultureInfo.InvariantCulture;
+            }
+            else
+            {
+                if (cultureInfo.NumberFormat.NumberDecimalSeparator.Equals(",") && separator.Equals(','))
+                {
+                    throw new ArgumentException("Decimal separator cannot match the column separator");
+                }
+            }
+
+            using (StreamWriter csvFile = new StreamWriter(csvStream, encoding ?? Encoding.UTF8))
+            {
+                if (dataFrame != null)
+                {
+                    var columnNames = dataFrame.Columns.GetColumnNames();
+
+                    if (header)
+                    {
+                        var headerColumns = string.Join(separator.ToString(), columnNames);
+                        csvFile.WriteLine(headerColumns);
+                    }
+
+                    var currPosition = csvStream.Position;
+
+                    foreach (var row in dataFrame.Rows)
+                    {
+                        var record = new StringBuilder();
+
+                        bool firstRow = true;
+                        foreach (var cell in row)
+                        {
+                            if (!firstRow)
+                            {
+                                record.Append(separator);
+                            }
+                            else
+                            {
+                                firstRow = false;
+                            }
+
+                            Type t = cell?.GetType();
+
+                            if (t == typeof(decimal))
+                            {
+                                record.Append(((decimal)cell).ToString(cultureInfo));
```

Comment:
I'm wondering if this overload is better here than calling `ToString(cultureInfo)`: https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.appendformat?view=netcore-3.1#System_Text_StringBuilder_AppendFormat_System_IFormatProvider_System_String_System_Object_System_Object_System_Object_